delta-lake/delta-33x/pom.xml

```diff
@@ -0,0 +1,96 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright (c) 2025 NVIDIA CORPORATION.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.nvidia</groupId>
+        <artifactId>rapids-4-spark-jdk-profiles_2.12</artifactId>
+        <version>25.06.0-SNAPSHOT</version>
+        <relativePath>../../jdk-profiles/pom.xml</relativePath>
+    </parent>
+
+    <artifactId>rapids-4-spark-delta-33x_2.12</artifactId>
+    <name>RAPIDS Accelerator for Apache Spark Delta Lake 3.3.x Support</name>
+    <version>25.06.0-SNAPSHOT</version>
+
+    <properties>
+        <rapids.module>../delta-lake/delta-33x</rapids.module>
+        <rapids.compressed.artifact>false</rapids.compressed.artifact>
+        <rapids.default.jar.excludePattern>**/*</rapids.default.jar.excludePattern>
+        <rapids.shim.jar.phase>package</rapids.shim.jar.phase>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.roaringbitmap</groupId>
+            <artifactId>RoaringBitmap</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.nvidia</groupId>
+            <artifactId>rapids-4-spark-sql_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
+            <classifier>${spark.version.classifier}</classifier>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.delta</groupId>
+            <artifactId>delta-spark_${scala.binary.version}</artifactId>
+            <version>${delta33x.version}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-sql_${scala.binary.version}</artifactId>
```

Comment:
I wonder why this is the `compile` scope instead of `provided`. Do we not want to use a specific version of spark-sql where the plugin runs with at runtime?

Reply:
Probably this is out of scope as I see other modules have the same dependency as `compile`. I'm just curious why it is done this way.

Reply:
It's `compile` scope which means that it will be packaged in the final jar. If it was `provided` that would make it open to whatever was on the classpath at runtime on the target env

Reply:
Yes, and I thought that's what we would like to. Maybe this is a more complex question than I thought. Why do we have any spark dependency as the `compile` dependency? We seem to have them as `provided` in the [main pom](https://github.com/NVIDIA/spark-rapids/blob/branch-25.06/pom.xml#L1088-L1091).

Reply:
My bad. This dependency is actually inherited from the parent, which has this `spark-sql` as `provided`. It makes sense now.

Reply:
Sorry, I didn't answer your question when I answered here but I hope it's clear now. 
The said dependency is inherited from the parent pom where we define this in the dependency management where the scope comes from

Reply:
Inconsistency always tends to generate otherwise redundant questions. It would make sense to consistently put dependencies under dependencyManagement or have the convention to explicitly define the scope.